tests/spec/methods/spec-immersive.js

```diff
@@ -0,0 +1,46 @@
+function testImmersive() {
+
+  it('Should call setCustomDeviceState with the correct object when called from screen with zoneId', function(){
+    var payload = [{color: "#fff", zoneId: 1}];
+    var expected_data = {__AC_IMMERSIVE_STATE__: {1: {color: "#fff"}}};
+    var expected_params = {action: "set", key: "custom", value: expected_data};
+    airconsole.devices[0].location = LOCATION;
+    spyOn(AirConsole, 'postMessage_');
+    airconsole.setImmersiveState(payload);
+    // Check data
+    var actual_data = airconsole.getCustomDeviceState();
+    expect(actual_data).toEqual(expected_data);
+    expect(AirConsole.postMessage_).toHaveBeenCalledWith(expected_params);
+  });
+
+  it('Should call setCustomDeviceState with the correct object when called from screen with many zoneId', function(){
+    var payload = [{color: "#fff", zoneId: 1}, {color: "#f00", zoneId: 0}];
+    var expected_data = {__AC_IMMERSIVE_STATE__: {1: {color: "#fff"},0:{color: "#f00"}}};
+    var expected_params = {action: "set", key: "custom", value: expected_data};
+    airconsole.devices[0].location = LOCATION;
+    spyOn(AirConsole, 'postMessage_');
+    airconsole.setImmersiveState(payload);
+    // Check data
+    var actual_data = airconsole.getCustomDeviceState();
+    expect(actual_data).toEqual(expected_data);
+    expect(AirConsole.postMessage_).toHaveBeenCalledWith(expected_params);
+  });
+
+
+  it('Should call setCustomDeviceState with the correct object when called from device', function(){
+    airconsole.devices[AirConsole.SCREEN] = { "device": "screen",  location: LOCATION };
+    airconsole.devices[2]  = { "device": "unicorn", location: LOCATION };
+    airconsole.setActivePlayers(1);
+    airconsole.device_id = 2;
+    var payload = [{color: "#fff"}];
+    var expected_data = {__AC_IMMERSIVE_STATE__: {0: {color: "#fff"}}};
+    var expected_params = {action: "set", key: "custom", value: expected_data};
+    airconsole.devices[0].location = LOCATION;
+    spyOn(AirConsole, 'postMessage_');
+    airconsole.setImmersiveState(payload);
+    // Check data
+    var actual_data = airconsole.getCustomDeviceState();
```

Comment:
This test to me looks wrong.

The documentation in airconsole-1.9.0-alpha and the above test imply that this is the state of the game, so setCustomState should only be set on the Screen (device_id === 0), if anyone else where to try to set it, it should throw an error like we do in setActivePlayers.

Otherwise we will not have **a single game state**: as the screen as well as every controller has their own state, at minimum confusing developers (even if the implementation on the appengine platform side would only listen to changes to the CustomState of device_id 0)